packages/sdk-cli/src/models/localRelay/relayInfo.ts

```diff
@@ -0,0 +1,93 @@
+import { cwd } from 'process';
+import { join } from 'path';
+import { waitUntil } from 'async-wait-until';
+
+import { isInt, readJsonFile } from './util';
+import { RELAY_PKG_NAME, RELAY_PID_FILE_PATH } from './const';
+
+export type RelayInfo = { port: number; pid: number };
+
+export type ReadRelayInfoResult = RelayInfo | false;
+
+export async function readRelayInfo(): Promise<ReadRelayInfoResult> {
+  try {
+    // Error will be thrown and caught if readJsonFile returns undefined or anything else that can't be destructured
+    const { port, pid } = (await readJsonFile(
+      RELAY_PID_FILE_PATH,
+    )) as RelayInfo;
+
+    // [port, pid].every(...) doesn't pass Control Flow Analysis. I.e. TS won't know port & pid are numbers.
+    if (isInt(port) && isInt(pid)) {
+      return {
+        port: Number(port),
+        pid: Number(pid),
+      };
+    }
+  } catch (error) {
+    if ((error as NodeJS.ErrnoException).code !== 'ENOENT') {
+      console.error(error);
+    }
+  }
+
+  return false;
+}
+
+export async function pollRelayInfo(
+  timeout = 15000,
+  interval = 300,
+): Promise<ReadRelayInfoResult> {
+  let relayInfo: ReadRelayInfoResult;
+
+  await waitUntil(async () => Boolean((relayInfo = await readRelayInfo())), {
+    timeout,
+    intervalBetweenAttempts: interval,
+  });
+
+  return relayInfo!;
+}
+
+export async function isRelayPkgInstalled(): Promise<boolean> {
+  // package.json _must_ be at the project root, and all dependencies _must_ be declared in package.json,
+  // so this is standard way to check for installed dependencies.
+  try {
+    const { dependencies, devDependencies } = (await readJsonFile(
+      join(process.cwd(), 'package.json'),
```

Comment:
NIT: use define for `package.json`

Reply:
What is that?

Reply:
Just a constant.

Reply:
Yes, just a constant, sorry for the confusion, C habits 🙈

Reply:
https://github.com/Fitbit/developer-bridge/pull/176/commits/1b5288515886cabe2144f191fece5e902354df49

Reply:
`https://github.com/Fitbit/developer-bridge/pull/176/commits/e5a905b2808c14ccf33cb65dcf30872a3715e108`

Reply:
I was thinking of adding a constant for the `package.json` file name as it is used in multiple places, the path part was ok. Example:
```
const packageConfig = 'package.json' // or any other name you find appropriate
...
join(process.cwd(), packageConfig),
...
```

Reply:
Oh! `package.json` is the standard (and I think required?) config file for all Node.js projects, so it's not something that is expected to be changed/configurable in the future, and it's always at the root of projects

Reply:
I'll leave this unresolved and add a TODO to fix it later; rn have too much stuff going on and tbh I'm scared of touching any parts of the repo